docs/new-architecture-appendix.md

```diff
@@ -292,7 +307,69 @@ node node_modules/react-native/scripts/generate-specs-cli.js \
 
 In the above example, the code-gen script will generate several files: `MyLibSpecs.h` and `MyLibSpecs-generated.mm`, as well as a handful of `.h` and `.cpp` files, all located in the `ios` directory.
 
-## V. Note on Existing Apps
+## V. Codegen Naming Conventions
+
+### Native Components
+
+Every native component spec file name must follow the following format:
+
+```
+<YourComponentName>NativeComponent.[js|ts|tsx]
+```
+
+The spec file must contain the following statement:
+
+```
+export default (codegenNativeComponent<NativeProps>(
+  '<YourComponentName>',
+): HostComponent<NativeProps>);
+```
+
+The codegen will produce the following native interfaces:
+
+<Tabs>
+  <TabItem value="apple" label="Java" default>
```

Comment:
"apple" for "Java"? I think we have some constants for languages: https://github.com/facebook/react-native-website/blob/main/website/core/TabsConstants.js